src/audio.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2020  Joshua Minter & Patryk Seregiet
+ *
+ * This file is part of ReplaySorcery.
+ *
+ * ReplaySorcery is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ReplaySorcery is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ReplaySorcery.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "audio.h"
+#include "util/log.h"
+#include "util/memory.h"
+
+static void rsAudioReadSamples(RSAudio *audio);
+static void rsAudioPrepareEncoder(RSAudio *audio, const RSConfig* config);
+static void rsAudioEncoderPrepareFrame(RSAudioEncoder *audioenc);
+
+extern sig_atomic_t mainRunning;
+extern sig_atomic_t wantToSave;
+
+int rsAudioCreate(RSAudio *audio, RSConfig *config) {
+   uint8_t channels = config->audioChannels;
+   if (channels != 1 && channels != 2) {
+      rsLog("Only 1 or 2 channels are supported. Setting to 2");
+   }
+
+   int error;
+   const pa_sample_spec ss = {
```

Comment:
Styling: use camelCase, not snake_case

Reply:
That's PulseAudio's typedef

Reply:
Sorry my bad, was looking at it on my phone while on a bus and it looked like a variable name